lib/pengine/utils.c

```diff
@@ -2568,3 +2568,43 @@ pe__build_rsc_list(pe_working_set_t *data_set, const char *s) {
 
     return resources;
 }
+
+xmlNode *
+pe__failed_probe_for_rsc(xmlNode *failed_ops, const char *rsc_id, const char *node_id)
+{
+    for (xmlNode *xml_op = pcmk__xml_first_child(failed_ops); xml_op != NULL;
+         xml_op = pcmk__xml_next(xml_op)) {
+        const char *value = NULL;
+        char *op_id = NULL;
+
+        /* This resource operation is not a failed probe. */
+        if (!pcmk__str_eq(crm_element_value(xml_op, "failed-probe"), "true", pcmk__str_none)) {
```

Comment:
What I really wanted was some sort of crm_attr_is_true function, but if I'm just going to get rid of failed-probe everywhere and redo the check, then I won't need it.

Reply:
Yeah I'm always surprised we don't have crm_xml_add_bool()/crm_element_value_bool() functions already, but never bother creating them. :) With current naming it might be something like pcmk__xe_bool_attr().

I've also toyed with the idea of variadic functions for getting/setting multiple attributes of varying types in one call, like

    pcmk_xe_set_attrs(xml, attrname1, pcmk_t_int, 1,
                                           attrname2, pcmk_t_str, "something",
                                           attrname3, pcmk_t_bool, true,
                                   NULL);
    pcmk_xe_get_attrs(xml, attrname1, pcmk_t_int, &intval,
                                           // etc.
                                   NULL);